### PR TITLE
Add some basic integration tests

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
It’s unlikely that we’re going to have a working unified test suite before the beta release, so here are some very basic smoke tests just to give us a _little bit_ of confidence that things are kind of working and that we don’t introduce major regressions.

Would be good to have a way of separating these from the unit tests so that they don’t slow them down, but can figure that out later; I don’t have loads of time to spend on this at the moment.

## Release Notes

- **New Features**
	- Added a new submodule for enhanced testing capabilities.
	- Introduced a `Sandbox` enum for creating API keys in the Ably sandbox environment.
	- Implemented basic integration tests for the AblyChat functionality to ensure proper message handling.

- **Documentation**
	- Updated `CONTRIBUTING.md` with new setup instructions and testing guidelines.
	- Modified `.prettierignore` to exclude specific test directories from formatting.

- **Chores**
	- Enhanced GitHub Actions workflow to include submodule checkouts for improved build processes.

### CODE REVIEW

File: Tests/AblyChatTests/IntegrationTests.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+        // 7. Fetch historical messages from before subscribing, and check we get txMessageBeforeRxSubscribe
+        let rxMessagesBeforeSubscribing = try await rxMessageSubscription.getPreviousMessages(params: .init())
+        #expect(rxMessagesBeforeSubscribing.items.count == 1)
+        #expect(rxMessagesBeforeSubscribing.items[0] == txMessageBeforeRxSubscribe)
+
+        // 8. Detach the room
+        try await rxRoom.detach()
+    }
+}
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Tests/AblyChatTests/IntegrationTests.swift:58]: What is `withTimeout`?

File: Tests/AblyChatTests/IntegrationTests.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        realtimeOptions.environment = "sandbox"
+        realtimeOptions.clientId = UUID().uuidString
+
+        return ARTRealtime(options: realtimeOptions)
+    }
+
+    private static func createSandboxChatClient(apiKey: String) -> DefaultChatClient {
+        let realtime = createSandboxRealtime(apiKey: apiKey)
+        return DefaultChatClient(realtime: realtime, clientOptions: nil)
+    }
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Tests/AblyChatTests/IntegrationTests.swift:21]: I’ll think about cleanup later

File: Tests/AblyChatTests/IntegrationTests.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Tests/AblyChatTests/IntegrationTests.swift]: I’ve added a comment to think about timeouts; don't have time to spend on implementing that now

**Comment by lawrence-forooghian** [comment]
Something is intermittently failing; will investigate.

**Comment by lawrence-forooghian** [comment]
OK, ready now.

**Review by maratal — APPROVED** [review]
LGTM

**Pull request merged**